local_apps/challenge/routes/challengeRouter.js

```diff
@@ -1,35 +1,143 @@
-const shuffle = require('lodash/shuffle')
-const moment = require('moment')
-const express = require('express')
-const Sequelize = require('sequelize')
-const { logger } = require('../index')
-// const { Session, Question, Answer, Tag, Activity } = require('../models')
-const User = require('../../../users/models/user')
-const requiresLogin = require('../../../utils/routes').requiresLogin
-const { QUERY_FORMAT } = require('../config')
-
-const challengeRouter = express.Router()
+const { Fight, Challenge, Question } = require('../models');
+const shuffle = require('lodash/shuffle');
+const moment = require('moment');
+const express = require('express');
+const Sequelize = require('sequelize');
+const { logger } = require('../index');
+const User = require('../../../users/models/user');
+const challenge = require('../models/challenge');
+const requiresLogin = require('../../../utils/routes').requiresLogin;
+const { QUERY_FORMAT } = require('../config');
+
+const challengeRouter = express.Router();
 
 challengeRouter.get('/', async (req, res, next) => {
-  res.json({ message: 'hello' })
-})
+  res.json({ message: 'hello' });
+});
 
 challengeRouter.get('/users', requiresLogin, async (req, res, next) => {
-  const query = req.query.name
-  console.log(query)
+  const query = req.query.name;
+  console.log(query);
   if (!query || query.length < 3 || !QUERY_FORMAT.test(query)) {
-    return next({ status: 400 })
+    return next({ status: 400 });
   }
   const users = await User.findAll({
     attributes: ['firstName', 'lastName', 'username'],
     where: {
       [Sequelize.Op.and]: [
         { id: { [Sequelize.Op.ne]: req.session.user.id } },
-        { username: { [Sequelize.Op.like]: `%${query}%` } }
-      ]
-    }
-  })
-  res.json(users)
+        { username: { [Sequelize.Op.like]: `%${query}%` } },
+      ],
+    },
+  });
+  res.json(users);
+});
+
+challengeRouter.post('/add', async (req, res, next) => {
+  const { challenger, challenged } = req.body;
```

Comment:
A user should not POST his own `id` as the challenger id. Instead, get the challenger id from `req.session.user.id`.

Reply:
I will change that as last thing jucăușe it is casier to test like that.